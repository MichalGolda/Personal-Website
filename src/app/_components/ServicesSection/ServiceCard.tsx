import Image from "next/image";

export type ServiceCardProps = {
  title: string;
  content: string;
  iconSrc: string;
};

const ServiceCard: React.FC<ServiceCardProps> = ({
  title,
  content,
  iconSrc,
  ...props
}) => {
  return (
    <div className="flex flex-col gap-y-4" {...props}>
      <div className="flex flex-row gap-x-4 md:flex-col md:gap-y-2 lg:flex-row md:items-start lg:items-center items-center">
        <Image
          src={iconSrc}
          alt={title}
          width={64}
          height={64}
          className="w-[48px] lg:w-[64px]"
        />
        <h4 className="font-bold text-lg md:text-xl lg:text-2xl text-secondary">
          {title}
        </h4>
      </div>
      <p className="text-body text-sm md:text-base">{content}</p>
      <span className="block bg-primary w-32 h-0.5"></span>
    </div>
  );
};

export default ServiceCard;
