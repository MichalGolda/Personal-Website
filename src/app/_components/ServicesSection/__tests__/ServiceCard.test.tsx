import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import ServiceCard from "../ServiceCard";

describe("ServiceCard component", () => {
  it("renders without errors", () => {
    const { container } = render(
      <ServiceCard
        title="test title"
        content="test content"
        icon={<>test icon</>}
      />
    );

    const title = screen.getByText("test title");
    const content = screen.getByText("test content");
    const icon = screen.getByText("test icon");

    expect(title).toBeInTheDocument();
    expect(title.tagName).toBe("H4");

    expect(content).toBeInTheDocument();
    expect(content.tagName).toBe("P");

    expect(icon).toBeInTheDocument();
  });

  it("renders unchanged", () => {
    const { container } = render(
      <ServiceCard
        title="test title"
        content="test content"
        icon={<>test icon</>}
      />
    );

    expect(container).toMatchSnapshot();
  });
});
