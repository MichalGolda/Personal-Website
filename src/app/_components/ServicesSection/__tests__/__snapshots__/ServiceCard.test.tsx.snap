// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceCard component renders unchanged 1`] = `
<div>
  <div
    class="flex flex-col gap-y-4"
  >
    <div
      class="flex flex-row gap-x-4 md:flex-col md:gap-y-2 lg:flex-row md:items-start lg:items-center items-center"
    >
      test icon
      <h4
        class="font-bold text-lg md:text-xl lg:text-2xl text-secondary"
      >
        test title
      </h4>
    </div>
    <p
      class="text-body text-sm md:text-base"
    >
      test content
    </p>
    <span
      class="block bg-primary w-32 h-0.5"
    />
  </div>
</div>
`;
